datalad/support/annexrepo.py

```diff
@@ -114,6 +115,10 @@ def __init__(self, path, url=None, runner=None,
           short description that humans can use to identify the
           repository/location, e.g. "Precious data on my laptop"
         """
+
+        if git_opts or annex_opts or annex_init_opts:
+            raise NotImplementedError("TODO")
```

Comment:
This seems excessive and will prevent us from starting to pass these options in new implementations. Warning should be enough.